### PR TITLE
Feature share table view controller

### PR DESCRIPTION
**Pull request by onevcat**

### CODE REVIEW

File: LineSDK/LineSDK/SharingUI/ShareTargetSelectingViewController.swift [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+        }
+        guard range.column == columnIndex else {
+            return
+        }
+        let indexPaths = (range.startIndex..<range.endIndex).map { IndexPath(row: $0, section: 0) }
+        tableView.insertRows(at: indexPaths, with: .none)
+    }
+
+    deinit {
+        print("Deinit: \(self)")
```
Comment (eJamesLin) [inline LineSDK/LineSDK/SharingUI/ShareTargetSelectingViewController.swift]: we might need to use `Log.print()` to identify it is print by SDK?

Comment (onevcat) [reply LineSDK/LineSDK/SharingUI/ShareTargetSelectingViewController.swift]: It is only in development and we need to remove these kind of things later when we done.

File: LineSDK/LineSDK/Utils/Helpers.swift [hunk 2]
```diff
@@ -97,6 +97,20 @@ extension UIViewController {
     }
 }
 
+extension UIViewController {
+    var expectedSearchBarHeight: CGFloat {
+        if #available(iOS 11.0, *) {
+            if UIApplication.shared.keyWindow?.safeAreaInsets.top == 20 {
+                return 44
+            } else {
+                return 54
```
Comment (eJamesLin) [inline LineSDK/LineSDK/Utils/Helpers.swift]: we might could remove the magic number by using `edgesForExtendedLayout = []`

Comment (onevcat) [reply LineSDK/LineSDK/Utils/Helpers.swift]: It seems that the embedded table view in the view controller fails to get the correct search bar height value. That is the reason why I hard coded them here. But we may try again to see whether `edgesForExtendedLayout` can work or not.

**Pull request merged**